WaveTournament/index.js

```diff
@@ -154,4 +160,26 @@ socket.onmessage = event => {
 			
 		}
 	}
+	if(!scoreVisibleTemp) {
+		if(chatLen != data.tourney.manager.chat.length) {
+			// There's new chats that haven't been updated
+			
+			if(chatLen == 0 || (chatLen > 0 && chatLen > data.tourney.manager.chat.length)) {
```

Comment:
Can you please explain this logic?

Reply:
```js
if(chatLen != data.tourney.manager.chat.length) {
```
This was made so the chat won't be updated unless necessary (when there's new chat available).

> So for example if we have
> ```
> A: some chat
> B: some chat
> ```
> and then there's a new chat
> ```
> A: some chat
> B: some chat
> C: some chat
> ```
> we only need to add the latest chat that we haven't add to our design (in this case C) instead of redoing it from the start (render from A to C).

`chatLen` stores how many chats there is. This logic is used when:
- There's a new chat, so `chatLen > data.tourney.manager.chat.length`
- You're changing room, so `chatLen != data.tourney.manager.chat.length` (There might be a edge case when the chats of a new room is the same as an old room though).

```js
if(chatLen == 0 || (chatLen > 0 && chatLen > data.tourney.manager.chat.length)) {
```
`chatLen == 0` will be true if it's our first time opening the design.
`(chatLen > 0 && chatLen > data.tourney.manager.chat.length)`  will be true if we enter other tournament lobby matches (ex.: from match Player 1 vs Player 2 to match Player 3 vs Player 4)
This conditional will redo the chat insertion from the very bottom (if we refer back to the example above, it'll try adding the chat element back from A to C)